phys/module_mp_rcon.F

```diff
@@ -0,0 +1,6217 @@
+!..
+!..            pres_Pa = Pressure in Pascals
+!..            w_non   = mixing ratio (non-dimensional = kg/kg)
+!..
+      IMPLICIT NONE
+
+      real:: pres_Pa, w_non
+      real:: p, RR, ES, ESLN
+
+!+---+
+
+      p = pres_Pa
+      RR=w_non+1e-8
+      ES=P*RR/(.622+RR)
+      ESLN=LOG(ES)
+      T_Dew=(35.86*ESLN-4947.2325)/(ESLN-23.6837)
+      return
+      end
+!
+!+---+-----------------------------------------------------------------+
+!
+      real function theta_wetb(thetae_K)
+!..
+!..              Eqn below was gotten from polynomial fit to data in
+!..              Smithsonian Meteorological Tables showing Theta-e
+!..              and Theta-w
+!..
+      IMPLICIT NONE
+
+      real:: thetae_K
+      real:: x, answer
+
+!+---+
+
+      real*8 c(0:6), d(0:6)
+      data c/-1.00922292e-10, -1.47945344e-8, -1.7303757e-6             &
+     &      ,-0.00012709,      1.15849867e-6, -3.518296861e-9           &
+     &      ,3.5741522e-12/
+      data d/0.00000000,   -3.5223513e-10, -5.7250807e-8                &
+     &     ,-5.83975422e-6, 4.72445163e-8, -1.13402845e-10              &
+     &     ,8.729580402e-14/
+
+      x=min(475.0,thetae_K)
+           
+      if( x .le. 335.5 ) then
+         answer = c(0)+x*(c(1)+x*(c(2)+x*(c(3)+x*(c(4)+x*(c(5)+         &
+     &            x*c(6) )))))
+      else
+         answer = d(0)+x*(d(1)+x*(d(2)+x*(d(3)+x*(d(4)+x*(d(5)+         &
+     &            x*d(6) )))))
+      endif
+
+      theta_wetb = answer + 273.15
+
+      return
+      end
+!
+!+---+-----------------------------------------------------------------+
+!
+      real function compT_fr_The(thelcl_K,pres_Pa)
+!..
+!..            pres_Pa = Pressure in Pascals
+!..            thelcl  = Theta-e at LCL (units in Kelvin)
+!..
+!..            Temperature (K) is returned given Theta-e at LCL
+!..            and a pressure.  This describes a moist-adiabat.
+!..            This temperature is the parcel temp at level Pres
+!..            along moist adiabat described by theta-e.
+!..
+      IMPLICIT NONE
+
+      real:: thelcl_K, pres_Pa
+      real:: guess, epsilon, w1, w2, tenu, tenup, cor, thwlcl_K
+      integer:: iter
+
+!+---+
+
+      guess= (thelcl_K - 0.5 * ( max(thelcl_K-270., 0.))**1.05)         &
+     &     * (pres_Pa/100000.0)**.2
+      epsilon=0.01
+      do iter=1,100
+         w1 = rslf(pres_Pa,guess)
+         w2 = rslf(pres_Pa,guess+1.)
+         tenu = theta_e(pres_Pa,guess,w1,guess)
+         tenup = theta_e(pres_Pa,guess+1,w2,guess+1.)
+         cor = (thelcl_K - tenu) / (tenup - tenu)
+         guess = guess + cor
+         if( (cor.lt.epsilon) .and. (-cor.lt.epsilon) ) then
+            compT_fr_The=guess
+            return
+         endif
+      enddo
+!     print*, '  convergence not reached '
+      thwlcl_K=theta_wetb(thelcl_K)
+      compT_fr_The = thwlcl_K*((pres_Pa/100000.0)**0.286)
+
+      return
+      end
+
+!+---+-----------------------------------------------------------------+
```

Comment:
I see that you removed your diagnostic arrays entirely. If you think you may want them in the future, an alternative would have been to keep them in this module and make them optional arguments which would make them easier to restore, but it would add local memory and computations unless you comment them out or use PRESENT checks.